abm/connector.py

```diff
@@ -89,12 +76,31 @@ def __del__(self):
         self.conf_file.close()
 
     '''
-    Translate the name of the temporary file in the host to the name of the same file
-    in the container.
-    For instance, it the path is '/tmp/tmp12345', return '/local/tmp12345'.
+    Remove metadata columns, if such exists, from "CATALOG" lines returned by an Airbyte read operation.
+    For instance, if a line is:
+        {'name': 'stream_name', 'json_schema': {'type': 'object', 'properties': {'_airbyte_stream_name_hashid': {'type': 'string'}, '_airbyte_ab_id': {'type': 'string'},
+        'dob': {'type': 'string'}, '_airbyte_normalized_at': {'type': 'string', 'format': 'date-time', 'airbyte_type': 'timestamp_without_timezone'},
+         'name': {'type': 'string'}, '_airbyte_emitted_at': {'type': 'string', 'format': 'date-time', 'airbyte_type': 'timestamp_with_timezone'}}}}
+    extract:
+        {'name': 'stream_name', 'json_schema': {'type': 'object', 'properties': {'dob': {'type': 'string'},'name': {'type': 'string'}}}}
+
+    These metadata columns are added in the normalization process.
+    ref:  https://docs.airbyte.com/understanding-airbyte/basic-normalization
     '''
-    def name_in_container(self, path):
-        return path.replace(self.workdir, MOUNTDIR, 1)
+    def remove_metadata_columns(self, line_dict):
+        catalog_streams = line_dict['catalog']['streams']
+        stream_name = self.get_stream_name()
+        for stream in catalog_streams:
+            # remove metadata columns for a specific stream (table) if such
+            # is provided
+            if stream_name != "" and stream['name'] != stream_name:
```

Comment:
consider replacing this line with:
```
if stream_name and stream['name'] != stream_name:
```

Reply:
Maybe there is no need to keep entries for all other streams (other than the stream you are really interested in)